reference.py

```diff
@@ -134,18 +135,31 @@ def create_outputs(input_priv_keys: List[Tuple[ECKey, bool]], input_hash: bytes,
 
     outputs = []
     for B_scan, B_m_values in silent_payment_groups.items():
-        k = 0
         ecdh_shared_secret = input_hash * a_sum * B_scan
 
-        # Order doesn't matter for creating/finding the outputs. However, different orderings
-        # may produce different generated outputs, if sending to multiple silent payment addresses belong to the
-        # same sender but different labels
-        for B_m in B_m_values:
-            t_k = TaggedHash("BIP0352/SharedSecret", ecdh_shared_secret.get_bytes(False) + ser_uint32(k))
-            P_km = B_m + t_k * G
-            outputs.append(P_km.get_bytes().hex())
-            k += 1
-    return outputs
+        # Order doesn't matter for creating/finding the outputs. However, different orderings of the recipient addresses
+        # will produce different generated outputs if sending to multiple silent payment addresses belong to the
+        # same sender but with different labels. To account for this in testing, set `perumute=True`. This will generate
+        # all possible sets of outputs (based on all possible orderings of recipient addresses).
+        #
+        # NOTE: THIS IS A TEST ONLY FEATURE AND NOT RELEVANT FOR NORMAL SENDING/RECEIVING
```

Comment:
Given that's a reference implementation that's not optimal to include test only code.

Reply:
I don't see an issue so long as its documented. I considered a few alternatives, but this seemed the simplest approach. Do you have another approach in mind?